be/src/connector/hive_connector.cpp

```diff
@@ -686,6 +686,16 @@ Status HiveDataSource::_init_scanner(RuntimeState* state) {
                 std::make_shared<TDeletionVectorDescriptor>(scan_range.deletion_vector_descriptor);
     }
 
+    if (dynamic_cast<const IcebergTableDescriptor*>(_hive_table)) {
+        auto tbl = dynamic_cast<const IcebergTableDescriptor*>(_hive_table);
+        scanner_params.lake_schema = tbl->get_iceberg_schema();
+    }
+
```

Comment:
I see this code if for iceberg, and i check this function not work well on iceberg, could you split into two?

Reply:
Yes, this code segment was accidentally deleted in PR #51050 so I add it back in this PR